synthetics/test_conversions.go

```diff
@@ -143,11 +143,41 @@ func testURLToMapSlice(obj *synthetics.V202101beta1UrlTest) []map[string]interfa
 	}}
 }
 
+//nolint: gocyclo, gocognit
 func testHealthSettingsToMapSlice(obj *synthetics.V202101beta1HealthSettings) []map[string]interface{} {
 	if obj == nil {
 		return nil
 	}
 
+	if obj.LatencyCritical == nil &&
+		obj.LatencyWarning == nil &&
+		obj.PacketLossCritical == nil &&
+		obj.PacketLossWarning == nil &&
+		obj.JitterCritical == nil &&
+		obj.JitterWarning == nil &&
+		obj.HttpLatencyCritical == nil &&
+		obj.HttpLatencyWarning == nil &&
+		obj.HttpValidCodes == nil &&
+		obj.DnsValidCodes == nil {
+		return nil
+	}
+
+	// Kentik API sets 0 values for nil fields
+	// Necessary conversion to nil, so Terraform configuration
+	// matches with actual state on the server
+	if *obj.LatencyCritical == 0 &&
+		*obj.LatencyWarning == 0 &&
```

Comment:
```suggestion
		obj.GetLatencyWarning() == 0 &&
```
Please use getters here to avoid the possibility of nil pointer dereference.

Reply:
Addressed: 0e8ee0d8730ca72fcd78be1b5f7aa7a7f5c0e604